utils/index.d.ts

```diff
@@ -25,6 +25,14 @@ export declare class AzExtTreeDataProvider implements TreeDataProvider<AzExtTree
      */
     public constructor(rootTreeItem: AzExtParentTreeItem, loadMoreCommandId: string);
 
+    /**
+     * Allows the data provider to listen to the {@link TreeView.onDidExpandElement} and {@link TreeView.onDidCollapseElement} events.
+     *
+     * This will ensure that the {@link AzExtParentTreeItem.collapsibleState} property is updated for the element.
+     * @param treeView The {@link TreeView} associated with this data provider
+     */
+    public setupCollapsibleStateListeners(treeView: TreeView<AzExtTreeItem>): Disposable;
```

Comment:
So is the issue that VSCode does not update the `collapsibleState` property on the `TreeItem`, when the tree item gets expanded by the user?

Reply:
Correct